src/Host/Broker/Impl/Security/SecurityManager.cs

```diff
@@ -96,10 +101,25 @@ public class SecurityManager {
                 _logger.LogTrace(Resources.Trace_LogOnSuccess, context.Username);
                 winIdentity = new WindowsIdentity(token);
 
-                StringBuilder profileDir = new StringBuilder(NativeMethods.MAX_PATH * 2);
+                StringBuilder profileDir;
                 bool profileExists = false;
+#if DEBUG
+                CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+#else
+                CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromSeconds(10));
+#endif
+
                 _logger.LogTrace(Resources.Trace_UserProfileCreation, context.Username);
-                error = CreateProfile(user.ToString(), domain.ToString(), context.Password, out profileDir, out profileExists);
+                JArray result = CreateProfileAsync(user.ToString(), domain.ToString(), context.Password, cts.Token).GetAwaiter().GetResult();
```

Comment:
Rather than blocking waiting for result, let's make this method async - it's caller is already async.